packages/block-library/src/button/edit.js

```diff
@@ -124,19 +165,37 @@ class ButtonEdit extends Component {
 								} }
 							/>
 						</PanelColorSettings>
+						<PanelBody title={ __( 'Link Settings' ) }>
+							<TextControl
+								label={ __( 'Link Rel' ) }
+								value={ rel || '' }
+								onChange={ this.onSetLinkRel }
+							/>
+						</PanelBody>
 					</InspectorControls>
 				</div>
 				{ isSelected && (
-					<form
-						className="block-library-button__inline-link"
-						onSubmit={ ( event ) => event.preventDefault() }>
-						<Dashicon icon="admin-links" />
-						<URLInput
-							value={ url }
-							onChange={ ( value ) => setAttributes( { url: value } ) }
-						/>
-						<IconButton icon="editor-break" label={ __( 'Apply' ) } type="submit" />
-					</form>
+					<URLPopover
+						position={ PopoverPosition }
+						renderSettings={ () => (
+							<ToggleControl
+								label={ __( 'Open in New Tab' ) }
+								onChange={ this.onSetNewTab }
+								checked={ linkTarget === '_blank' }
+							/>
+						) }
+					>
+						<form
+							className="block-library-button__inline-link"
+							onSubmit={ ( event ) => event.preventDefault() }>
+							<Dashicon icon="admin-links" />
+							<URLInput
+								value={ url }
+								onChange={ ( value ) => setAttributes( { url: value } ) }
+							/>
+							<IconButton icon="editor-break" label={ __( 'Apply' ) } type="submit" />
```

Comment:
The apply button here doesn't do seem to do anything.

My feeling is that it'd be good to make this consistent with the link format (i.e. the link UI in the paragraph block). It'd be great if it could toggle between an 'editing' state and a 'viewing' state a bit like that does:
https://github.com/WordPress/gutenberg/blob/master/packages/format-library/src/link/inline.js#L268-L281

It's a fair amount of boilerplate code to do that, unfortunately. Hopefully that's something in the codebase that can be improved in the future.

Reply:
Interesting, thank you for the reference @talldan, I tried implementing it but it looks like this will need a bit more time so I'll pick up again a bit later and try again ❤️

Reply:
Hey @nfmohit-wpmudev - I used the button block today and was reminded of this PR 😄 

Let me know if you need any assistance on this.

Reply:
Hey @talldan ❤️ 
Thank you so much for your assistance with this, I'm ever grateful.

As per our discussion in DM, I have removed the Apply button for now so that we can go forward with it. I'm trying to come up with the edit state link viewer in a new PR.

Thank you ❤️